Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -1,62 +1,115 @@
 package com.gildedrose;
 
+import static com.gildedrose.ItemNameValues.AGED_BRIE;
+import static com.gildedrose.ItemNameValues.BACKSTAGE_PASSES;
+import static com.gildedrose.ItemNameValues.SULFURAS;
+
 class GildedRose {
+
     Item[] items;
 
     public GildedRose(Item[] items) {
         this.items = items;
     }
 
+    public void updateQualityV2(){
+        for (Item item : items) {
+            final String itemName = item.name;
+
+            switch (itemName){
```

Comment:
aínda que aos Clean Coders máis puros non lles gusta o switch, o mellora é moi grande, moi ben.

Reply:
Engadino porque me parece que queda o código máis sinxelo de entender.

De todas formas para engadir o novo tipo de arquivo teño que cambiar a lóxica xa que agora teño que comprobar se o nome comeza cun determinado texto